src/encoded/static/components/filegallery.js

```diff
@@ -3748,6 +3744,226 @@ FileGalleryRendererComponent.contextTypes = {
+        const referenceEpigenomeInfo = [relatedDataset.assay_term_name];
+        if (relatedDataset.target) {
+            referenceEpigenomeInfo.push(relatedDataset.target.label);
+        }
+        return referenceEpigenomeInfo.join(', ');
+    },
+
+    ReplicationTimingSeries: (relatedDataset) => {
+        let phases = [];
+        if (relatedDataset.replicates && relatedDataset.replicates.length > 0) {
+            const biosamples = relatedDataset.replicates.map((replicate) => replicate.library && replicate.library.biosample);
+            phases = _.chain(biosamples.map((biosample) => biosample.phase)).compact().uniq().value();
+        }
+        return phases.join(', ');
+    },
+
+    TreatmentConcentrationSeries: (relatedDataset) => {
+        let concentration = [];
+        if (relatedDataset.replicates && relatedDataset.replicates.length > 0) {
+            const biosamples = relatedDataset.replicates.map((replicate) => replicate.library && replicate.library.biosample);
+            if (biosamples.length > 0 && biosamples[0].treatments && biosamples[0].treatments.length > 0 && biosamples[0].treatments[0].amount) {
+                const treatment = biosamples[0].treatments[0];
+                concentration = `${treatment.treatment_term_name} ${treatment.amount} ${treatment.amount_units}`;
+            }
+        }
+        return concentration;
+    },
+
+    TreatmentTimeSeries: (relatedDataset) => {
+        let durations = [];
+        if (relatedDataset.replicates && relatedDataset.replicates.length > 0) {
+            const biosamples = relatedDataset.replicates.map((replicate) => replicate.library && replicate.library.biosample);
+            durations.push(biosamples.map((biosample) => {
+                if (biosample.treatments && biosample.treatments.length > 0) {
+                    const treatment = biosamples[0].treatments[0];
+                    durations = `${treatment.treatment_term_name} ${treatment.duration} ${treatment.duration_units}${treatment.duration > 1 ? 's' : ''}`;
+                }
+                return null;
+            }));
+        }
+        return durations;
+    },
+};
+
+
+/**
+ * Display the title portion for a file details table section.
+ */
+const TitleMainContent = ({
+    context,
+    analysis,
+    title,
+    totalFiles,
+    isDownloadable,
+    batchDownloadController,
+}) => {
+    // Based on the @type of the dataset, get a composer function appropriate for it to generate
+    // a subtitle, passing it the related datasets if it exists. Any dataset objects without a
+    // composer function get rendered normally.
+    let subtitle;
+    const type = context['@type'][0];
+    if (context.related_datasets && context.related_datasets.length > 0 && analysis) {
+        const relatedDataset = context.related_datasets.find((dataset) => analysis.datasets.includes(dataset['@id']));
+        if (relatedDataset) {
+            const composer = subtitleComposers[type];
```

Comment:
Where is the logic for reading series without a composer?

Reply:
Without a composer, you get an empty `subtitle`, and the title gets renders like it does in production today. That applies to all non-Series objects too.

Reply:
thanks.